### PR TITLE
[NOC-276] example of changes required for SSOT db connection

### PR DESCRIPTION
**Pull request by jonathan-longe**
# Switch MSSQL Enterprise Database with Postgres SSOT

After exploring this option, I don't recommend this approach. Microsoft does not make it easy to switch from MSSQL to Postgres.  It feels like a game of whack-a-mole -- solve one problem and another crops up.  To date, I have been unable to get the JobBoardContext and the SSOTContext running simultaneously.  If we're forced to use this approach, we might consider a tool like Bablefish to emulate MSSQL using Postgres.

### CODE REVIEW

File: src/WorkBC.Data/JobBoardContext.cs [hunk 1]
```diff
@@ -139,9 +139,16 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+            /*    modelBuilder.Entity<JobSeeker>()
+                   .HasIndex(x => new { x.AccountStatus, x.LastName, x.FirstName })
+                   .IncludeProperties(modelBuilder.Entity<>(x => new { x.Email }));
+
+
+                *
+                * NpgsqlIndexBuilderExtensions.IncludeProperties(
+       modelBuilder.Entity<Blog>().HasIndex(f => f.Name),
+       f => f.Description);
+
```
Comment (jonathan-longe) [inline src/WorkBC.Data/JobBoardContext.cs:151]: If we add a Postgres DB, the JobBoardContext would need to be modified to clarify which database is being used for the connection.

File: src/WorkBC.Data/SsotContext.cs [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+{
+    public partial class SsotContext : DbContext
+    {
+        private string ssotConnectionString;
+        
+        public SsotContext(DbContextOptions<SsotContext> options, IConfiguration configuration) 
+            : base(options)
+        {
+            ssotConnectionString = configuration.GetConnectionString("SsotConnection");
+        }
```
Comment (jonathan-longe) [inline src/WorkBC.Data/SsotContext.cs:15]: I've created a SsotContext as switching databases is more complicated than changing the connection string for the 'EnterpriseContext'.

File: src/WorkBC.Data/SsotContext.cs [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            optionsBuilder.UseNpgsql(ssotConnectionString);
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<CareerProfile>(entity =>
+            {
+                entity.HasKey(e => e.CareerProfileId);
+
+                entity.ToTable("edm_careerprofile");
```
Comment (jonathan-longe) [inline src/WorkBC.Data/SsotContext.cs:32]: Postgres table and column names are case-insensitive so every table name, will need to be modified here to show the new Postgres table name.

File: src/WorkBC.Data/SsotContext.cs [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+                entity.HasIndex(e => e.NocId).IsUnique();
+                
+                entity.Property(e => e.CareerProfileId).HasColumnName("careerprofileid");
+
+                entity.Property(e => e.AnnualSalarySource).HasMaxLength(255);
```
Comment (jonathan-longe) [inline src/WorkBC.Data/SsotContext.cs:38]: Similarly, each column name will need to have a `.HasColumnName()` added to convert the TitleCase to lowercase.

File: src/WorkBC.Web/Controllers/CareerProfilesController.cs [hunk 5]
```diff
@@ -23,11 +23,11 @@ namespace WorkBC.Web.Controllers
     public class CareerProfilesController : ControllerBase
     {
         private readonly JobBoardContext _context;
-        private readonly EnterpriseContext _enterpriseContext;
+        private readonly SsotContext _enterpriseContext;
         private readonly IConfiguration _configuration;
         private readonly IGeocodingService _geocodingService;
 
-        public CareerProfilesController(JobBoardContext context, EnterpriseContext enterpriseContext,
+        public CareerProfilesController(JobBoardContext context, SsotContext enterpriseContext,
```
Comment (jonathan-longe) [inline src/WorkBC.Web/Controllers/CareerProfilesController.cs:30]: The changes to the Controller are relatively straight-forward.  The `EnterpriseContext` needs to the switched to the `SsotContext`.

**Comment by infojunkie** [comment]
I agree. If the switch to another database is that tricky, we might as well do the "right" thing and connect to SSOT via API. Feel free to close this PR and add your notes to the ticket please.

**Pull request closed without merging**